server/src/main/java/org/opensearch/cluster/metadata/AutoExpandReplicas.java

```diff
@@ -134,6 +134,14 @@ int getMaxReplicas(int numDataNodes) {
         return Math.min(maxReplicas, numDataNodes - 1);
     }
 
+    public int getMaxReplicas() {
+        return maxReplicas;
```

Comment:
Should we use AutoExpandReplica::getDesiredNumberOfReplicas?

Reply:
The `getDesiredNumberOfReplicas` is used for updating the replicaCount of the index which has `auto_expand` setting enabled. We here want to validate the maxCap(constant) which is set while defining the index setting.